Performing C SOURCE FILE Test CMAKE_HAVE_LIBC_PTHREAD failed with the following output:
Change Dir: /home/marcussss1/2022_2_MidnightSun/build/CMakeFiles/CMakeTmp

Run Build Command(s):/usr/bin/make -f Makefile cmTC_9fa0f/fast && /usr/bin/make  -f CMakeFiles/cmTC_9fa0f.dir/build.make CMakeFiles/cmTC_9fa0f.dir/build
make[1]: вход в каталог «/home/marcussss1/2022_2_MidnightSun/build/CMakeFiles/CMakeTmp»
Building C object CMakeFiles/cmTC_9fa0f.dir/src.c.o
/usr/bin/cc -DCMAKE_HAVE_LIBC_PTHREAD   -o CMakeFiles/cmTC_9fa0f.dir/src.c.o -c /home/marcussss1/2022_2_MidnightSun/build/CMakeFiles/CMakeTmp/src.c
Linking C executable cmTC_9fa0f
/home/marcussss1/.local/lib/python3.9/site-packages/cmake/data/bin/cmake -E cmake_link_script CMakeFiles/cmTC_9fa0f.dir/link.txt --verbose=1
/usr/bin/cc -rdynamic CMakeFiles/cmTC_9fa0f.dir/src.c.o -o cmTC_9fa0f 
/usr/bin/ld: CMakeFiles/cmTC_9fa0f.dir/src.c.o: в функции «main»:
src.c:(.text+0x46): неопределённая ссылка на «pthread_create»
/usr/bin/ld: src.c:(.text+0x52): неопределённая ссылка на «pthread_detach»
/usr/bin/ld: src.c:(.text+0x5e): неопределённая ссылка на «pthread_cancel»
/usr/bin/ld: src.c:(.text+0x6f): неопределённая ссылка на «pthread_join»
collect2: error: ld returned 1 exit status
make[1]: *** [CMakeFiles/cmTC_9fa0f.dir/build.make:99: cmTC_9fa0f] Ошибка 1
make[1]: выход из каталога «/home/marcussss1/2022_2_MidnightSun/build/CMakeFiles/CMakeTmp»
make: *** [Makefile:127: cmTC_9fa0f/fast] Ошибка 2


Source file was:
#include <pthread.h>

static void* test_func(void* data)
{
  return data;
}

int main(void)
{
  pthread_t thread;
  pthread_create(&thread, NULL, test_func, NULL);
  pthread_detach(thread);
  pthread_cancel(thread);
  pthread_join(thread, NULL);
  pthread_atfork(NULL, NULL, NULL);
  pthread_exit(NULL);

  return 0;
}

Determining if the function pthread_create exists in the pthreads failed with the following output:
Change Dir: /home/marcussss1/2022_2_MidnightSun/build/CMakeFiles/CMakeTmp

Run Build Command(s):/usr/bin/make -f Makefile cmTC_3b9da/fast && /usr/bin/make  -f CMakeFiles/cmTC_3b9da.dir/build.make CMakeFiles/cmTC_3b9da.dir/build
make[1]: вход в каталог «/home/marcussss1/2022_2_MidnightSun/build/CMakeFiles/CMakeTmp»
Building C object CMakeFiles/cmTC_3b9da.dir/CheckFunctionExists.c.o
/usr/bin/cc   -DCHECK_FUNCTION_EXISTS=pthread_create -o CMakeFiles/cmTC_3b9da.dir/CheckFunctionExists.c.o -c /home/marcussss1/.local/lib/python3.9/site-packages/cmake/data/share/cmake-3.24/Modules/CheckFunctionExists.c
Linking C executable cmTC_3b9da
/home/marcussss1/.local/lib/python3.9/site-packages/cmake/data/bin/cmake -E cmake_link_script CMakeFiles/cmTC_3b9da.dir/link.txt --verbose=1
/usr/bin/cc  -DCHECK_FUNCTION_EXISTS=pthread_create -rdynamic CMakeFiles/cmTC_3b9da.dir/CheckFunctionExists.c.o -o cmTC_3b9da  -lpthreads 
/usr/bin/ld: невозможно найти -lpthreads
collect2: error: ld returned 1 exit status
make[1]: *** [CMakeFiles/cmTC_3b9da.dir/build.make:99: cmTC_3b9da] Ошибка 1
make[1]: выход из каталога «/home/marcussss1/2022_2_MidnightSun/build/CMakeFiles/CMakeTmp»
make: *** [Makefile:127: cmTC_3b9da/fast] Ошибка 2



